source/includes/_endpoint-dataset.md

```diff
@@ -540,6 +540,85 @@ PATCH the "expression" attribute to modify. An empty "expression" object, like
 
 ##### Main deck
 
+#### Primary key
+
+`/datasets/{dataset_id}/pk/`
+
+Setting a primary key on a dataset causes updates (particularly streamed
+updates) mentioning existing rows to be updated instead of new rows being
+inserted.  A primary key can only be set on a column that has structured data,
```

Comment:
Users don't know what "structured" means here. Say "of text or numeric type" instead.